testing/upload_validation.ts

```diff
@@ -947,11 +947,13 @@ const genericObjectSchema: z.ZodTypeAny = z.lazy(() =>
         isValidSchema: (schema: Schema & ObjectSchemaProperty) => boolean,
         invalidSchemaMessage: string,
       ) {
-        if (schema[propertyKey]) {
+        const propertyValueRaw = schema[propertyKey];
+        if (propertyValueRaw) {
           const propertyValue =
-            typeof schema[propertyKey] === 'string'
-              ? (schema[propertyKey] as string)
-              : (schema[propertyKey] as ObjectProperty).value;
+            typeof propertyValueRaw === 'string'
+              ? propertyValueRaw
+              : (propertyValueRaw as PropertyIdentifierDetails)?.property;
```

Comment:
The type safety here could be a bit better, which will fix the need for the cast. (Trying to avoid casting except when really necessary, like when zod gives you an `any` value or something.) As the number of contributors to the SDK grows, people won't have as much context to know where in the code to proactively look to fix stuff and we'll need to rely on TS and tests to catch potential edge cases, so casting too much will mask a class of bugs.

If we declare an intermediate type like:

```typescript
type SchemaPathProperties = Pick<GenericObjectSchema, 'titleProperty' | 'linkProperty' | 'imageProperty' | 'descriptionProperty'>;
```

Then you can type `propertyKey` as `propertyKey: keyof SchemaPathProperties`, and now TS knows that `propertyValueRaw` is `PropertyIdentifier<string>` and the cast isn't necessary.

Reply:
ah very good call, will add that